dev/encoding/vqgan-jax-encoding-yfcc100m-streaming.ipynb

```diff
@@ -0,0 +1,604 @@
+{
```

Comment:
<p>If I use the local dataset (with <code>streaming=False</code>),  it takes about 15mn 55 sec</p><p> </p><p>It takes only 1mn 46s if I use the version suggested above with:</p><pre class="ql-syntax" spellcheck="false">prepared_dataset_non_batched = dataset.map(prepare_data_non_batched,
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;remove_columns=remove_columns,
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;num_proc=20)
</pre><p> </p><p>I took advantage to remove the columns not needed:</p><pre class="ql-syntax" spellcheck="false">remove_columns = list(dataset.column_names)
remove_columns.remove('key')
remove_columns
</pre><p> </p><p>Note that num_proc may be able to be increased but I had errors when it was too much (maybe also because I kept all the values of datasets).</p><p>I'm not exactly sure how data is cached but somehow my drive went only from 3.6TB&nbsp;to 3.7TB.</p><p> </p><p>Once it's cached, the same function takes only 9 seconds!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/borisdayma/dalle-mini/pull/84/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Reply:
<p>I forgot to remove the columns, good catch! And <code>num_proc</code>  seems to help a lot, the contents of each zip file must be copied across subprocesses, maybe, and loaded in parallel? That's a great performance improvement!</p>

Reply:
<p>Also, very surprised about the cache behavior. Not sure what's going on there.</p>